unit-tests/live/d400/test-hdr.py

```diff
@@ -0,0 +1,638 @@
+                        frame_gain = out_depth_frame.get_frame_metadata(rs.frame_metadata_value.gain_level)
+                        seq_id = out_depth_frame.get_frame_metadata(rs.frame_metadata_value.sequence_id)
+
+                        if seq_id == 0:
+                            test.check(frame_exposure == first_exposure)
+                            test.check(frame_gain == first_gain)
+                        else:
+                            test.check(frame_exposure == second_exposure)
+                            test.check(frame_gain == second_gain)
+
+                depth_sensor.set_option(rs.option.hdr_enabled, 0)  # disable hdr before next tests
+        else:
+            log.e("No device was found - skipping test")
+
+# CHECKING HDR AFTER PIPE RESTART
+
+with test.closure("HDR Running - restart hdr at restream", "[hdr][live][using_pipeline]"):
+    ctx = rs.context()
+    device = test.find_first_device_or_exit()
+    if ctx:
+        devices_list = ctx.query_devices()
+        device_count = devices_list.size()
+        if devices_list.size() != 0:
+            depth_sensor = device.first_depth_sensor()
+            if depth_sensor and depth_sensor.supports(rs.option.hdr_enabled):
+
+                cfg = rs.config()
+                cfg.enable_stream(rs.stream.depth)
+                pipe = rs.pipeline()
+                pipe.start(cfg)
+
+                depth_sensor.set_option(rs.option.hdr_enabled, 1)
+                test.check(depth_sensor.get_option(rs.option.hdr_enabled) == 1)
+
+                for i in range(10):
+                    data = pipe.wait_for_frames()
+
+                test.check(depth_sensor.get_option(rs.option.hdr_enabled) == 1)
+                pipe.stop()
+
+                pipe.start(cfg)
+                test.check(depth_sensor.get_option(rs.option.hdr_enabled) == 1)
+                pipe.stop()
+
+                depth_sensor.set_option(rs.option.hdr_enabled, 0)  # disable hdr before next tests
+        else:
+            log.e("No device was found - skipping test")
+
+
+# helper method
+# checks that the frames resulting from the hdr_merge processing block are generated
+# from frames streamed within this method (and not using old frames)
+# returned value: False if some metadata could not be read - otherwise, True
+def check_hdr_frame_counter(pipe, num_of_frames, merging_filter):
+    min_counter = -1
+    min_counter_set = False
+    for i in range(num_of_frames):
+        data = pipe.wait_for_frames()
+
+        # get depth frame data
+        depth_frame = data.get_depth_frame()
+        if not depth_frame.supports_frame_metadata(rs.frame_metadata_value.sequence_id):
+            log.e("Frame metadata not supported")
+            return False
+
+        depth_counter = depth_frame.get_frame_metadata(rs.frame_metadata_value.frame_counter)
+
+        if depth_counter < min_counter:  # in case of frame counter reset during stream this might happen
+            log.e("Frame counter reset!")
```

Comment:
So you decided to fail the test on frame counter reset?
How will this stabilize the test?

Reply:
this does not fail the test, just print that it happened.
I added later `min_counter_set = False` which causes the next condition to happen and reset min_counter

Reply:
Yes it does..
https://github.com/IntelRealSense/librealsense/blob/4673a37d981164af8eeb8e296e430fc1427e008d/unit-tests/py/rspy/log.py#L165

Reply:
oh, that's new to me, I didn't see it failed the test even though it was printed...
what should I replace it with? log.d ?

Reply:
Could be because you didn't had
`print_results_and_exit`
You can replace with log.d, yes

Reply:
I don't understand 1 this, once we set min counter we go in a loop and make sure the next frame counter is higher than the first frame counter?
So we only check that the sensor does not reset itself and that is exactly what we relax now?

Can you explain (maybe ask Remi) what is the actual test here?

Reply:
Yes, I went with the idea of the comment above this function, that what we test here is that the frames we get from  hdr_merge are generated from new frames generated in this function and not from older ones (with lower frame counter). So I don't think that a frame counter reset should fail this test..?

Reply:
OK, so we are saying we make sure the counter keep increasing and not stay the same as it was before.
It it's smaller we will write in the log and reinitialize our min_counter, right?